runtime/lua/vim/lsp/handlers.lua

```diff
@@ -97,6 +97,18 @@ M['window/showMessageRequest'] = function(_, _, params)
   end
 end
 
+--@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#client_registerCapability
+M['client/registerCapability'] = function(_, _, _, client_id)
+  local warning = "The language server %s triggers a registerCapability "..
+                  "handler despite dynamicRegistration set to false. "..
+                  "Report upstream, this warning is harmless"
+  local client = vim.lsp.get_client_by_id(client_id)
+  local client_name = client and client.name or string.format("id=%d", client_id)
+  warning = string.format(warning, client_name)
```

Comment:
Reusing the name `warning` hurts readibility (its a small snippet but in the grand scheme of things). I usually use XXX_tpl to signify it's a `template` string => warning_tpl and warning.

Reply:
Done